src/Sentry.Hangfire/SentryJobFilter.cs

```diff
@@ -0,0 +1,87 @@
+using Hangfire.Server;
+using Sentry.Extensibility;
+
+namespace Sentry.Hangfire;
+
+internal class SentryJobFilter : IServerFilter
+{
+    internal const string SentryMonitorSlugKey = "SentryMonitorSlug";
+    internal const string SentryCheckInIdKey = "SentryCheckInIdKey";
+
+    private readonly IHub _hub;
+    private readonly IDiagnosticLogger? _logger;
+
+    public SentryJobFilter() : this(() => HubAdapter.Instance)
+    { }
+
+    internal SentryJobFilter(Func<IHub> hubAccessor)
+    {
+        _hub = hubAccessor.Invoke();
+        _logger = _hub.GetSentryOptions()?.DiagnosticLogger;
+    }
+
+    public void OnPerforming(PerformingContext context)
+    {
+        OnPerformingInternal(context, GetJobParameter, SetJobParameter);
+    }
+
+    public void OnPerformed(PerformedContext context)
+    {
+        OnPerformedInternal(context, GetJobParameter, HasException);
+    }
+
+    internal void OnPerformingInternal(PerformingContext? context,
+        Func<PerformContext?, string, object?> getJobParameterFunc,
+        Action<PerformContext?, string, SentryId> setJobParameter)
+    {
+        if (getJobParameterFunc(context, SentryMonitorSlugKey) is not string monitorSlug)
+        {
+            var jobName = context?.BackgroundJob.Job.ToString();
```

Comment:
Does Job have a Name property or something? Relying on `ToString` here is less optimal if that's the case becauase it could change

Reply:
`Job` has a `type` and `method` and `ToString` concatenates this to `Type.Method`. 
Users opt-in to have their jobs monitored when they add `.UseSentry`. If the job does not have a monitor slug the warning tells them which specific type and method is missing the attribute.

Reply:
Not sure I follow. My point is that we're relying on ToString where it might be safer to rely on a different property which indicates why it exists. ToString can become a longer thing that includes other stuff, later for other reasons